README.md

```diff
@@ -35,3 +35,95 @@ console.log(binaryPacket.toString('hex'))
 
 const jsonPacket = packet.deserializeIlpPayment(binaryPacket)
 ```
+
+### Serialize/deserialize ILQP Quote Requests/Responses
+
+#### QuoteLiquidityRequest
+
+```js
+const packet = require('ilp-packet')
+
+const binary = packet.serializeIlqpLiquidityRequest({
+  requestId: 'c217449a-69e9-4a08-9575-6e7eafa818c0',
+  destinationAccount: 'example.nexus.bob',
```

Comment:
> The messages should not be so ambiguous that you need to check 4 cases

I didn't say you should **check** four cases (in your implementation.) I said you should **test** them (in development.) You do the same thing in all cases. The only reason I suggested that people should test it is precisely to make sure that they **didn't** add any unnecessary and counterproductive `if` statements in their implementation.

> to figure out if the response tells you what you wanted to know.

The response always tells you what you wanted to know. What are you talking about?

> I want a liquidity curve to a specific account. Anything less specific than that is meaningless to me and I will need to discard it anyway. In this case, I would prefer for the connector to give me an error saying "I don't know".

Any liquidity curve to a prefix of that account is also a liquidity curve to that account. When quoting you can only return a prefix if the curve applies to all accounts under that prefix.

> I want a liquidity curve to a ledger. Again, if you don't know the curve all the way to that ledger, I'd probably prefer an "I don't know".

If they don't know the curve to that prefix, that means either there is no path to get to that prefix or there isn't a single curve that applies to the whole prefix. And yes, the response in that case is an error.

> I want a liquidity curve to as close to a given address as possible.

Quoting for a liquidity curve always gives you the curve between you and the destination. There is no such thing as a curve "as close to a given address as possible." Either there is a path, in which case you'll get back the curve, or you quoted for something broader than a ledger or account, in which case you'll get an error, or there is no path in which case you'll also get an error.

> Based on these, I don't see a use case for the connector responding for a prefix that is something other than what I asked for.

The use case is:

`example.ledger1.alice` wants to send 100 payments to individual accounts `example.blah.dest.bob1`, `example.blah.dest.bob2`, `example.blah.dest.bob3` etc. So to start, she asks for a quote for destination account `example.blah.dest.bob1`. She receives a response with destination prefix `example.blah.dest.`. For the 99 other payments she does not need to quote again.

This is going to be an extremely common case (basically all recurring payments) because we do things like append unique IDs to ILP addresses.

Note that if there truly is a different liquidity curve for each account, then alice would get the response with destinationPrefix `example.blah.dest.bob1`. Which is still great, because if she is now sending multiple payments to `example.blah.dest.bob1.tx1`, `tx2`, etc., she can still reuse that curve.

So it's super-useful when you query for a curve to get back the minimum prefix that the curve applies to.

> Regarding the field names, I think the best options are either to split out the messages into two so they can be precise

There is only one case: I'm quoting for a destination and get back a curve and a prefix to tell me what other destinations that curve applies to. So I'm strongly against having two messages.

> or to make the field name something that is more general and includes both accounts and ledgers like "address" or "destination".

I think `destinationPrefix` on the return value makes sense. But I could see why you would rename the field on the request `destination`. The reason I called it `destinationAccount` is because you're not supposed to know other people's ledger prefixes. People give you accounts to send to, they shouldn't really give you prefixes ever. Anytime you are quoting for a prefix, you've probably poorly designed your protocol, so I didn't want the field name to indicate that we're supporting/encouraging that behavior, although there is no reason we need to break it either.